pkg/image/api/helper_test.go

```diff
@@ -122,10 +122,16 @@ func TestParseImageStreamTagName(t *testing.T) {
 }
 
 func TestParseDockerImageReference(t *testing.T) {
+	parserFuncs := map[string]func(spec string) (DockerImageReference, error){
+		"ParseDockerImageReference":              ParseDockerImageReference,
+		"ParseMultiSegmentsDockerImageReference": ParseMultiSegmentsDockerImageReference,
+	}
+
 	testCases := []struct {
 		From                               string
 		Registry, Namespace, Name, Tag, ID string
 		Err                                bool
+		MSegErr                            bool
```

Comment:
I agree with Jordan - more explicit tests are better.  Use a common core array and add variant cases.